src/test/java/org/javarosa/core/model/SelectChoiceTest.java

```diff
@@ -119,4 +123,103 @@ public void getChild_updates_whenChoicesAreFromRepeat() throws IOException {
         scenario.answer("/data/repeat[0]/special-property", "changed");
         assertThat(scenario.choicesOf("/data/select").get(0).getChild("special-property"), is("changed"));
     }
+
+    @Test
+    public void getChild_throwsException_whenCalledOnAChoiceFromStaticSelect() throws IOException {
```

Comment:
This will result in a crash on "static" selects in Collect that add the `map` appearance right? I'm still unhappy with the fact that it doesn't work (although I understand that it's an existing issue) but I think we should handle this differently with one of these options:

1. `getChild`/`getAdditionalChildren` throws a checked exception that Collect can anticipate (rather than a `RuntimeException`) and handle with an error message
1. `getChild`/`getAdditionalChildren` return null/empty and Collect disables the "Select place" button for "static" selects (and shows an error on the widget)

Reply:
Shoot, I meant to call this out but forgot. My reasoning was that once you're here it's too late and that it should be a client's responsibility to avoid calling these methods in contexts where they're not applicable. These methods only make sense when there's an itemset. So e.g. the `map` appearance which relies on them only makes sense when there's an itemset.

BUT. Do clients know whether they're dealing with an inline or itemset-based select?

> Collect disables the "Select place" button for "static" selects 

Is that possible? If so, then the runtime exception continues to make sense to me.

Reply:
> Is that possible? If so, then the runtime exception continues to make sense to me.

Huh, good question. I'm not sure. I'm still unclear on why a "static" select can't have children - it seems incredibly confusing, and I think the situation will be exacerbated by the `map` appearance (people will assume it works with "static" selects). If that's something we intend to "fix" then I think returning null/empty alone is the best course of action for the moment.